src/EhrgoHealth.WebService/Models/AllergyIntolerance.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Rest;
+
+namespace EhrgoHealth.WebService.Models
+{
+    public class AllergyIntolerance
+    {
+        //Dictonary looks up possible allergies for a given medication. Example allergyLookup<Medication, List<Codes>>
+        private Dictionary<string, List<string>> allergyLookup = new Dictionary<string, List<string>>();
+        private FhirClient fhirClient = null;
+
+
+        public AllergyIntolerance(string fhirServer)
+        {
+            if (fhirServer == null || fhirServer == String.Empty)
+                throw new Exception("Invalid URL passed to AllergyIntolerance Constructor");
+
+            fhirClient = new FhirClient(fhirServer);
+
+            //We do not use a database for the webservice. We will hardcode a handful of allergies and their intolerances
+            allergyLookup.Add("hydrocodone", new List<string>() { "Z88.5" });
+        }
+
+        //The controller is expected to receive two pieces of data from an EHR.
+        //The List of medications they currently know their patient is taking
+        //The patient's ID from FHIR
+        /// <summary>
+        /// The controller is expected to receive two data structures from an EHR.
+        /// </summary>       
+        /// <param name="patientID"> The patient's ID from FHIR</param>
+        /// <param name="medications"> The List of medications they currently know their patient is taking</param>
+        public List<string> GetListOfMedicationAllergies(uint patientID, List<string> medications)
+        {
+            //ToDo: Discuss with team about returning a dictionary or wrapper class, so that we can inform the EHR
+            //      on both the medication conflict and the allergy code.
+
+
+            //First let us fetch the known allergies of the patient.
+            List<string> returnedAllergies = new List<string>();
+            Dictionary<string, Boolean> lookupPatientsKnownAllergies = GetPatientsKnownAllergies(patientID);
+            if (lookupPatientsKnownAllergies == null)
+            {
+                //There are no records on the FHIR server of this patient having any allergies.
+                return returnedAllergies;
+            }         
+
+            ////Now we retrieve the list of known allergies their medications can trigger
+            List<string> listOfAllergicMedications = DetermineListOfAllergicMedications(medications, lookupPatientsKnownAllergies);

```

Comment:
you could have just written `return DetermineListOfAllergicMedications(medications, lookupPatientsKnownAllergies);